dante/osomerank/topic_diversity.py

```diff
@@ -41,21 +43,24 @@ def load_td_data():
         if not os.path.exists(cached_model_dir):
             logger.warning("No cached model found! Retrieving from S3.")
             fetchfroms3(prefix, cache_path)
+
         # XXX: do not calculate probabilities?
         TD_MODEL = BERTopic.load(cached_model_dir)
         logger.info(f"Loaded BERTopic model from: {cached_model_dir}")
-    except Exception as e:
-        logger.error(f"Failed to load BERTopic model from {cached_model_dir}: {e}")
 
-    try:
         json_fn = config.get("AUDIENCE_DIVERSITY", "topic_diversity_json")
         cached_json_path = os.path.join(cache_path, json_fn)
         with open(cached_json_path) as ff:
             TD_DATA = json.load(ff)
-    except Exception as e:
+
+    except (configparser.NoSectionError, OSError) as e:
         logger.error(
-            f"Failed to load topic diversity data from {cached_json_path}: {e}"
+            f"Failed to load BERTopic model or topic diversity data from {cached_json_path}: {e}"
         )
+        raise
```

Comment:
Same as before, this except block should only wrap the line where you do `config.get(...)`